app/scripts/metamask-controller.js

```diff
@@ -121,6 +122,16 @@ module.exports = class MetamaskController extends EventEmitter {
         .then((newState) => { cb(null, newState) })
         .catch((reason) => { cb(reason) })
       },
+      importAccountWithStrategy: (strategy, args, cb) => {
+        accountImporter.importAccount(strategy, args)
+        .then((privateKey) => {
+          return keyringController.addNewKeyring('Simple Key Pair', [ privateKey ])
```

Comment:
This action is only really useful for importing individual accounts representing a single keypair.  We still have the `addNewKeyring` action for creating an arbitrary keyring (like HD, or whatever else we add).